test/test_site/testUtil/test.js

```diff
@@ -18,16 +21,22 @@ for (let i = 0; i < expectedPaths.length; i += 1) {
     throw new Error('Different files built');
   }
 
-  // compare html files only
-  if (path.parse(actualFilePath).ext === '.html') {
+  // compare html files and site data
+  if (path.parse(actualFilePath).ext === '.html' || actualFilePath.includes('siteData.json')) {
     const resolvedExpectedFilePath = path.resolve('./expected', expectedFilePath);
     const resolvedActualFilePath = path.resolve('./_site', actualFilePath);
     const expected = fs.readFileSync(resolvedExpectedFilePath, 'utf8');
     const actual = fs.readFileSync(resolvedActualFilePath, 'utf8');
-    try {
-      diffHtml(expected, actual);
-    } catch (err) {
-      throw new Error(`${err.message} in ${expectedFilePath}`);
+    if (actualFilePath.includes('siteData.json')) {
+      if (!_.isEqual(JSON.parse(expected), JSON.parse(actual))) {
+        throw new Error('Site data does not match with the expected file.');
+      }
+    } else {
+      try {
+        diffHtml(expected, actual);
+      } catch (err) {
+        throw new Error(`${err.message} in ${expectedFilePath}`);
+      }
```

Comment:
Hmm, since it's quite distinct, this may be clearer:

```js
const parsed = path.parse(actualFilePath);
if (parsed.ext === '.html') {
  // compare html files
  // ...
} else if (parsed.base === 'siteData.json') {
  // compare site data
  // ...
}
```